src/dodal/plans/scanspec.py

```diff
@@ -0,0 +1,64 @@
+import operator
+from collections.abc import Mapping
+from functools import reduce
+from typing import Annotated, Any
+
+import bluesky.plans as bp
+from bluesky.protocols import Movable, Readable
+from cycler import Cycler, cycler
+from pydantic import Field, validate_call
+from scanspec.specs import Spec
+
+from dodal.common import MsgGenerator
+from dodal.plan_stubs import attach_data_session_metadata_decorator
+
+
+@attach_data_session_metadata_decorator()
+@validate_call(config={"arbitrary_types_allowed": True})
+def spec_scan(
+    detectors: Annotated[
+        set[Readable],
+        Field(
+            description="Set of readable devices, will take a reading at each point, \
+            in addition to any Movables in the Spec",
+        ),
+    ],
+    spec: Annotated[
+        Spec[Movable],  # type: ignore
```

Comment:
@coretl @callumforrester do you have any idea why this is currently required? Previously we had a Spec[str] in blueapi and this did not cause issues- here in dodal, it seems we cannot have a Spec[str] or Spec[Movable].

`"__getitem__" method not defined on type "type"Pylance[reportIndexIssue](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportIndexIssue)`

This is an issue in Pylance in the IDE and also in the type-checking step of the CI job.

Reply:
This was due to the discriminated_union decorator's type hinting not being adequate.
This change is therefore blocked until ScanSpec 0.7.3 can be released, which is blocked until time is spent to handle dataclasses.